app/src/main/java/com/keylesspalace/tusky/TuskyApplication.kt

```diff
@@ -65,15 +68,20 @@ class TuskyApplication : Application(), HasAndroidInjector {
 
         AppInjector.init(this)
 
-        val preferences = PreferenceManager.getDefaultSharedPreferences(this)
+        // Migrate shared preference keys and defaults from version to version. The last
+        // version that did not have a SCHEMA_VERSION was 97, so that's the default.
+        val oldVersion = sharedPreferences.getInt(PrefKeys.SCHEMA_VERSION, 97)
```

Comment:
(It's actually 100. But I guess that is not so important.)

Reply:
Wait a sec, that is not working. We release the nightly version on every commit to develop, and it has completely different version codes. Needs another solution @nikclayton

Reply:
We could still "merge" it with the migration code - or at least something similar. It has a manual "counter".

Reply:
Ok, probably only something similar. As the `AppModule#providesDatabase` does not really have an explicit location where it is called and where we could add a `doPreferenceUpgrade()`.